src/api/controllers/socialAuth.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-else-return */
+import model from '../models';
+import generateToken from '../../helpers/tokens/generate.token';
+
+const { User } = model;
+
+/**
+ *
+ *
+ * @class socialLogin
+ */
+class socialLogin {
+  /**
+   * signup via facebook
+   * login  via facebook
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async socialFacebookAuth(req, res) {
+    let findIfExist = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!findIfExist) {
+      findIfExist = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+    }
+    const registerUser = { username: req.user.displayName, uniqueId: req.user.id };
```

Comment:
@Ramadhan0 `registerUser` isn't a good name for this variable. Favour nouns for variable names and verbs for method/function names.